core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/SegmentIndexFileStore.java

```diff
@@ -338,6 +338,24 @@ private MergedBlockIndex readMergeBlockIndex(ThriftReader thriftReader) throws I
     });
   }
 
+  /**
+   * List all the index files of the segment.
+   *
+   * @param carbonFile directory
+   */
+  public static void getCarbonIndexFilesRecursively(CarbonFile carbonFile,
```

Comment:
Why not return the list instead of passing it as a parameter?

Reply:
Its a recursive call, and it adds file to the passed list if it finds any.